rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp

```diff
@@ -686,14 +670,208 @@ error_code cellSysutilAvc2SetSpeakerMuting(u8 muting)
 	return CELL_OK;
 }
 
+error_code cellSysutilAvc2Load_shared(SceNpMatching2ContextId ctx_id, u32 container, vm::ptr<CellSysutilAvc2Callback> callback_func, vm::ptr<void> user_data, vm::cptr<CellSysutilAvc2InitParam> init_param)
+{
+	if (!init_param || !init_param->avc_init_param_version ||
+	    !(init_param->avc_init_param_version == 100 ||
+	     init_param->avc_init_param_version == 110 ||
+	     init_param->avc_init_param_version == 120 ||
+	     init_param->avc_init_param_version == 130 ||
+	     init_param->avc_init_param_version == 140)
+	)
+	{
+		return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+	}
+
+	auto& settings = g_fxo->get<avc2_settings>();
+
+	switch (init_param->media_type)
+	{
+	case CELL_SYSUTIL_AVC2_VOICE_CHAT:
+	{
+		if (init_param->max_players < 2 ||
+		    init_param->max_players > 64 ||
+		    init_param->spu_load_average > 100 ||
+		    init_param->voice_param.voice_quality != CELL_SYSUTIL_AVC2_VOICE_QUALITY_NORMAL ||
+		    init_param->voice_param.max_speakers == 0 ||
+		    init_param->voice_param.max_speakers > 16
+		)
+		{
+			return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+		}
+
+		u32 streaming_mode = init_param->direct_streaming_mode;
+
+		if (init_param->avc_init_param_version < 120)
+		{
+			if (init_param->avc_init_param_version >= 110)
```

Comment:
I think this can be reordered so this if would appear in an else path for example, it looks odd nested here.

Reply:
done